src/analytics/internal/send-to-server/send-to-server.js

```diff
@@ -101,11 +101,13 @@ class SendToServer {
     }
 
     async shouldTrack() {
+        const idDoNotTrackEnabled = window.navigator.doNotTrack
+
         const storage = await browser.storage.local.get({
             [SHOULD_TRACK]: SendToServer.DEF_TRACKING,
         })
 
-        return storage[SHOULD_TRACK]
+        return storage[SHOULD_TRACK] && !idDoNotTrackEnabled
```

Comment:
I would check DNT is set before getting the should track setting from storage. Then you can avoid the async stuff completely if it's turned on